### PR TITLE
(Publish TBD) Stripe Integration documentation

### PR DESCRIPTION
**Pull request by sjyothi54**
**Ticket:** https://issues.newrelic.com/browse/NR-156208
**Description:** Created Stripe integration documentation.
**Preview image**
screencapture-localhost-8000-docs-infrastructure-host-integrations-host-integrations-list-stripe-integration-2023-08-31-20_01_12

Please follow [conventional commit standards](https://www.conventionalcommits.org/en/v1.0.0/)
in your commit messages and pull request title.

## Give us some context

* What problems does this PR solve?
* Add any context that will help us review your changes such as testing notes,
  links to related docs, screenshots, etc.
* If your issue relates to an existing GitHub issue, please link to it.

### CODE REVIEW

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+* Follow our [guided install](https://one.newrelic.com/nr1-core?state=999174fa-ed8b-d707-af7b-8e4f8fd03cc1) to instrument your system with the infrastructure agent. Or if you prefer, you can install the infrastructure agent by following the docs.
+* By integrating this [infrastructure agent](https://docs.newrelic.com/docs/infrastructure/install-infrastructure-agent/get-started/install-infrastructure-agent/) in your system, one can able to 
+  know the CPU usage, Memory usage, Storage usage and Network traffic etc.
+ 
+## Install application specific New Relic agent
+
+* Follow the steps in our [guided install](https://one.newrelic.com/nr1-core?state=999174fa-ed8b-d707-af7b-8e4f8fd03cc1) for a quick and easy install process.
+
+## Install Browser monitoring agent
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx:34]:
```suggestion
## Install our browser monitoring agent
```

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+Our Browser monitoring agent reports application and browser-related metrics like AJAX performance, error rate, duration, core web vitals, and throughput.
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx:35]:
```suggestion
Our browser monitoring agent reports application and browser-related metrics like AJAX performance, error rate, duration, core web vitals, and throughput.
```

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+* To install the agent, Follow this steps to install [Browser Monitoring Agent](https://docs.newrelic.com/docs/browser/browser-monitoring/installation/install-browser-monitoring-agent/)
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx:36]:
```suggestion
See [Install the browser monitoring agent](/docs/browser/browser-monitoring/installation/install-browser-monitoring-agent/) to get started.
```

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+## Set up synthetic monitoring
+Our synthetic monitors test your app in different locations and capture server downtime. You can view this data in the Stripe dashboard.
+To set up synthetic monitors, follow the steps in our [guided install](https://onenr.io/0EjOXrnxgQ6)
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx:39]:
```suggestion
## Set up synthetic monitoring
Our synthetic monitors test your app in different locations and capture server downtime. You can view this data in the Stripe dashboard.

To set up synthetic monitoring, follow the steps in our [guided install](https://onenr.io/0EjOXrnxgQ6).
```

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+## Send metrics to New Relic
+Check the [Metric API documentation](https://docs.newrelic.com/docs/data-apis/ingest-apis/metric-api/report-metrics-metric-api/) for further details.
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx:42]:
```suggestion
Check the [Metric API documentation](/docs/data-apis/ingest-apis/metric-api/report-metrics-metric-api/) for further details.
```

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+To proceed further, you will need a New Relic license key. Kindly follow the [New Relic API key documentation](https://docs.newrelic.com/docs/apis/intro-apis/new-relic-api-keys/) to get your license key.
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx:44]:
```suggestion
To proceed further, you will need a New Relic license key. See [New Relic API Keys](/docs/apis/intro-apis/new-relic-api-keys/) to get your license key.
```

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+On a successful stripe payment, we receive a JSON from the stripe API service.
+We need to send these values to New Relic by formatting the JSON as per New Relic Metric API requirements.
+
+Below is the javascript code snippet which formats the data and sends it to New Relic metric API URL
+```
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx:49]:
```suggestion
On a successfulStripe payment, we receive a JSON from the Stripe API service. We need to send these values to New Relic by formatting the JSON as per New Relic metric API requirements.

Below is the JavaScript code snippet that formats the data and sends it to New Relic metric API URL:

```
```

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+    console.log('Metrics data sent successfully:', response.data);
+  })
+  .catch((error) => {
+    // Handle any errors that occur during the request
+    console.error('Error sending metrics data to New Relic:', error);
+  })
+}
+
+```
+**Note:** Change the sessionJSON according to your response and call this function where you are getting stripe metrics in your application.
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx:96]:
```suggestion
**Note:** Change the `sessionJSON` according to your response and call this function where you are getting Stripe metrics in your application.
```

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+```
+ sendDataToNewRelic (session)
+```
+
+## Monitor your application
+You can choose our pre-built dashboard template named `Stripe` to monitor your Stripe application metrics.
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx:102]:
```suggestion
You can choose our pre-built Stripe dashboard to monitor your application metrics. To set up this dashboard:
```

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+1. Go to **[one.newrelic.com](https://one.newrelic.com/)** and click on **+ Add data**.
+2.  Click on the **Dashboards** tab.
+3.  In the search box, type `Stripe`.
+4.  When you see our pre-build dashboard, click on it to install it in your account.
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx:106]: @sjyothi54 I wasn't able to find this dashboard. Can you provide a direct link to it? 
image

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+</figcaption>
+
+## Install the infrastructure agent [#infra]
+
+Before getting Stripe data into New Relic, first install our infrastructure agent, then you will be able to get details about your host.
+
+* Follow our [guided install](https://one.newrelic.com/nr1-core?state=999174fa-ed8b-d707-af7b-8e4f8fd03cc1) to instrument your system with the infrastructure agent. Or if you prefer, you can install the infrastructure agent by following the docs.
+* By integrating this [infrastructure agent](https://docs.newrelic.com/docs/infrastructure/install-infrastructure-agent/get-started/install-infrastructure-agent/) in your system, one can able to 
+  know the CPU usage, Memory usage, Storage usage and Network traffic etc.
+
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx:29]:
```suggestion
Before getting Stripe data into New Relic, first install our infrastructure agent, and then you will be able to get details about your host, such as CPU usage, memory usage, storage usage, network traffic, and more.

Follow our [guided install](https://one.newrelic.com/nr1-core?state=999174fa-ed8b-d707-af7b-8e4f8fd03cc1) to instrument your system with the infrastructure agent. Or if you prefer, you can manually install the infrastructure agent by following the [installation docs](/docs/infrastructure/install-infrastructure-agent/get-started/install-infrastructure-agent/). 
 
```

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+## Install the infrastructure agent [#infra]
+
+Before getting Stripe data into New Relic, first install our infrastructure agent, then you will be able to get details about your host.
+
+* Follow our [guided install](https://one.newrelic.com/nr1-core?state=999174fa-ed8b-d707-af7b-8e4f8fd03cc1) to instrument your system with the infrastructure agent. Or if you prefer, you can install the infrastructure agent by following the docs.
+* By integrating this [infrastructure agent](https://docs.newrelic.com/docs/infrastructure/install-infrastructure-agent/get-started/install-infrastructure-agent/) in your system, one can able to 
+  know the CPU usage, Memory usage, Storage usage and Network traffic etc.
+ 
+## Install application specific New Relic agent
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx:30]:
```suggestion
## Install an application-specific New Relic agent
```

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx [hunk 13]
```diff
@@ -0,0 +1,125 @@
...
+
+* Follow the steps in our [guided install](https://one.newrelic.com/nr1-core?state=999174fa-ed8b-d707-af7b-8e4f8fd03cc1) for a quick and easy install process.
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx:32]:
```suggestion
* Follow the steps in our [guided install](https://one.newrelic.com/nr1-core?state=999174fa-ed8b-d707-af7b-8e4f8fd03cc1) to install on of our agents based on your application type.
```

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx (hunk 7 above)
Comment (ally-sassman) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx:49]:
```suggestion
```javascript
```

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx [hunk 14]
```diff
@@ -0,0 +1,125 @@
...
+  })
+  .catch((error) => {
+    // Handle any errors that occur during the request
+    console.error('Error sending metrics data to New Relic:', error);
+  })
+}
+
+```
+**Note:** Change the sessionJSON according to your response and call this function where you are getting stripe metrics in your application.
+```
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx:97]:
```suggestion
```javascript
```

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx (hunk 10 above)
Comment (sjyothi54) [reply src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx:106]: Stripe dashboard:https://onenr.io/0ZQWYYE1rRW

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx [hunk 15]
```diff
@@ -0,0 +1,125 @@
...
+## Monitor your application
+You can choose our pre-built dashboard template named `Stripe` to monitor your Stripe application metrics.
+1. Go to **[one.newrelic.com](https://one.newrelic.com/)** and click on **+ Add data**.
+2.  Click on the **Dashboards** tab.
+3.  In the search box, type `Stripe`.
+4.  When you see our pre-build dashboard, click on it to install it in your account.
+
+Once your application is integrated by following the above steps, the dashboard should display metrics.
+
+To instrument the Stripe quickstart and to see metrics and alerts, you can also follow our [Stripe quickstart page](https://newrelic.com/instant-observability/stripe) by clicking on the **Install now** button.
```
[the remark this answers is not in the document]
Comment (sjyothi54) [reply src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx:110]: @ally-sassman  The quickstart isn't live yet

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx [hunk 16]
```diff
@@ -0,0 +1,123 @@
...
+## Monitor your application
+You can choose our pre-built Stripe dashboard to monitor your application metrics. To set up this dashboard:
+1. Go to **[one.newrelic.com](https://one.newrelic.com/)** and click on **+ Add data**.
+2.  Click on the **Dashboards** tab.
+3.  In the search box, type `Stripe`.
+4.  When you see our pre-build dashboard, click on it to install it in your account.
+
+Once your application is integrated by following the above steps, the dashboard should display metrics.
+
+To instrument the Stripe quickstart and to see metrics and alerts, you can also follow our [Stripe quickstart page](https://newrelic.com/instant-observability/stripe) by clicking on the **Install now** button.
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx:110]:
```suggestion
## Monitor your application
You can choose our pre-built Stripe dashboard to monitor your application metrics. To set up this dashboard, go to [https://onenr.io/0ZQWYYE1rRW] and follow the instructions there. Once installed, the dashboard should display metrics.
```

Thanks @sjyothi54 . I replaced this section with just a single link to go to the UI.

**Comment by ally-sassman** [comment]
@sjyothi54 Please post a @hero message in the #help-documentation Slack channel when this feature is live, and we will merge the PR.

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx [hunk 17]
```diff
@@ -0,0 +1,116 @@
+---
+title: Stripe integration
+tags:
+  - Stripe integration
+  - New Relic integration
+metaDescription: Install our Stripe dashboards and see your Stripe data in New Relic.  
+---
+import infrastructureStripeDashboard from 'images/infrastructure_screenshot-stripe-dashboard.webp'
+
+By utilizing New Relic Stripe dashboard, you gain the ability to monitor your application's transaction overview, transaction volume, revenue and transaction errors effortlessly. The Stripe dashboard is developed with our application specific monitoring agent, browser monitoring agent offering you a consolidated view of critical data, conveniently accessible in one central location.
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx:10]:
```suggestion
The New Relic Stripe dashboard provides a consolidated view of your application's transaction overview, transaction volume, revenue, and transaction errors. The dashboard is developed with our application-specific monitoring agent and browser monitoring agent, so you can easily monitor your application's performance in one central location.
```

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx [hunk 18]
```diff
@@ -0,0 +1,116 @@
...
+
+<img
+    title="Stripe"
+    alt="Stripe dashboard"
+    src={infrastructureStripeDashboard}
+/>
+
+<figcaption>
+After setting up our Stripe, you can install a dashboard for your Stripe metrics.
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx:19]:
```suggestion
After setting up monitoring for your Stripe app, you can view all your Stripe metrics in one dashboard.
```

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx [hunk 19]
```diff
@@ -0,0 +1,140 @@
...
+To set up synthetic monitors, follow the steps in our [guided install](https://onenr.io/0EjOXrnxgQ6)
+  </Step>
+
+    <Step>
+### Send metrics to New Relic [#send-metrics]
+
+Check the [Metric API documentation](/docs/data-apis/ingest-apis/metric-api/report-metrics-metric-api/) for further details.
+
+To proceed further, you will need a New Relic license key. See [New Relic API Keys](/docs/apis/intro-apis/new-relic-api-keys/) to get your license key.
+On a successful stripe payment, we receive a JSON from the stripe API service. We need to send these values to New Relic by formatting the JSON as per New Relic Metric API requirements.
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx:61]: @sjyothi54 Should this be renamed as an optional step to set up custom metrics? 
```suggestion
### (Optional) Send custom metrics to New Relic [#send-metrics]

If you need to customize the app metrics sent to New Relic, see [Metric API documentation](/docs/data-apis/ingest-apis/metric-api/report-metrics-metric-api/) to learn how. Once you've set up custom metrics, with a successful stripe payment, we receive a JSON from the stripe API service. We need to send these values to New Relic by formatting the JSON as per New Relic Metric API requirements.

If you're sending custom metrics, you'll need a New Relic license key. See [New Relic API Keys](/docs/apis/intro-apis/new-relic-api-keys/) to get your license key.
```

Comment (sjyothi54) [reply src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx:61]: @ally-sassman This step is essential, not optional.

**Comment by sjyothi54** [comment]
> @sjyothi54 Please post a @hero message in the #help-documentation Slack channel when this feature is live, and we will merge this PR.

@ally-sassman Once the Stripe quickstart is live, [this link](https://newrelic.com/instant-observability/stripe) will be functional. However, we need to ensure that the Stripe documentation is also live before adding the Stripe doc link to the Stripe quickstart. cc @rahulbasu-nr @mdumpati

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx [hunk 20]
```diff
@@ -0,0 +1,140 @@
...
+
+Below is the javascript code snippet which formats the data and sends it to New Relic metric API URL
+```javascript
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx:64]:
```suggestion
The following JavaScript code snippet formats the data and sends it to the New Relic metric API URL: 

```javascript
```

File: src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx [hunk 21]
```diff
@@ -0,0 +1,140 @@
...
+ sendDataToNewRelic (session)
+```
+
+      </Step>
+
+  <Step>
+### Install our Stripe monitoring dashboard [#add-dashboard]
+
+You can choose our pre-built Stripe dashboard to monitor your application metrics. To set up this dashboard, go to [https://onenr.io/0ZQWYYE1rRW] and follow the instructions there. Once installed, the dashboard should display metrics.
```
Comment (ally-sassman) [inline src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx:123]:
```suggestion
To set up our pre-built Stripe dashboard to monitor your application metrics, go to the [Stripe dashboard installation](https://onenr.io/0ZQWYYE1rRW) and follow the instructions. Once installed, the dashboard should display metrics.
```

**Review by ghost — APPROVED** [review]

**Comment by ally-sassman** [comment]
@sjyothi54 Sounds good, just let us know when [this link](https://onenr.io/0ZQWYYE1rRW) is live and we'll merge this PR.

**Comment by pkosanam** [comment]
Hi @ally-sassman , 

We are creating this documentation to attach it to the upcoming "Stripe" quickstart on https://newrelic.com/instant-observability platform.

Currently the quickstart PR https://github.com/newrelic/newrelic-quickstarts/pull/2004, is marked as draft as it need the documentation link to be attached to the quickstart.

Requesting you to kindly merge this PR, so that we can proceed further with the quickstart PR.

cc: @rahulbasu-nr @mdumpati

**Pull request merged**